server/auth/store.go

```diff
@@ -935,6 +939,8 @@ func NewAuthStore(lg *zap.Logger, be AuthBackend, tp TokenProvider, bcryptCost i
 
 	be.CreateAuthBuckets()
 	tx := be.BatchTx()
+	// We should call LockWithoutHook here, but the txPostLockHoos isn't set
```

Comment:
I would worry about introducing this assumption. It might be true now, but we should minimize number of logical dependencies that cannot be enforced in code.

Reply:
All operations outside the apply workflow are supposed to call `LockWithoutHook`, so does `NewAuthStore` . But it's still in progress of etcd startup/initialization (because it's called in [NewServer](https://github.com/etcd-io/etcd/blob/a08d479463960948d5d9c3508623ade915c6e66f/server/etcdserver/server.go#L373)), which means etcd hasn't started to serve any requests yet, so it definitely has no any impact on the apply workflow. 

From the implementation perspective, we don't even have an instance of [EtcdServer](https://github.com/etcd-io/etcd/blob/a08d479463960948d5d9c3508623ade915c6e66f/server/etcdserver/server.go#L202) yet, it isn't possible to call the [getTxPostLockHook](https://github.com/ahrtr/etcd/blob/2c961171cd07f7fba2da9fe41ca8507a81b9ec1b/server/etcdserver/server.go#L2332) 

So we should be good here.

Reply:
PTAL https://github.com/etcd-io/etcd/issues/13883
Looks like cache in AuthStore is not reset correctly. Depending on implementation of fix we might decide to call Recover during etcd runtime

Reply:
Please note that the logic (auth.Recover) is already there, see [server.go#L1012](https://github.com/etcd-io/etcd/blob/a08d479463960948d5d9c3508623ade915c6e66f/server/etcdserver/server.go#L1012). The issue [13883](https://github.com/etcd-io/etcd/issues/13883) is just related to `rangePermCache`, so it should have no any impact on this PR.